### PR TITLE
Improve testability, add phpstan

### PR DESCRIPTION
**Pull request by silvano-a**
This should make #4 easier.

### CODE REVIEW

File: src/Autodiscovery/AutodiscoveryLockResolver.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+final class AutodiscoveryLockResolver
+{
+    public const NOT_IN_LOCK = 'notInLock';
+    public const NOT_IN_AUTOLOAD = 'notInAutoload';
+    public function resolve(LaravelPackageManifest $manifest): Collection
+    {
+        $autoload = $manifest->collectManifestFromComposerAutoload();
+        $lock = $manifest->collectManifestFromLock();
+
+        return collect(
```
Comment (NiekKeijzer) [inline src/Autodiscovery/AutodiscoveryLockResolver.php:16]: The collection seems to be used as fancy associative array, maybe a value object could help here?

File: src/Commands/AutodiscoveryPackageLockVerify.php [hunk 2]
```diff
@@ -26,52 +25,54 @@ public function __construct(PackageManifest $manifest)
...
-                });
+            $this->displayNotInAutoloadMessages($errorMessages->get(AutodiscoveryLockResolver::NOT_IN_AUTOLOAD));
+            $this->displayNotInLockMessages($errorMessages->get(AutodiscoveryLockResolver::NOT_IN_LOCK));
 
-                throw new \Exception('A missmatch between the lock file and the autodiscovered packages was found.');
-            } else {
-                $this->info('The lock file is up to date with the autodiscovered packages.');
-            }
-            return self::SUCCESS;
+            throw new \Exception('A mismatch between the lock file and the autodiscovered packages was found.');
```
Comment (NiekKeijzer) [inline src/Commands/AutodiscoveryPackageLockVerify.php:49]: Generic exception is used for control flow, I think this should be moved outside of the `try` / `catch` and simply return `self::FAILURE`. Alternatively specific Exceptions could be used to provide better error messages

File: phpstan.neon.dist [hunk 3]
```diff
@@ -0,0 +1,4 @@
+parameters:
+    level: 5
```
[the remark this answers is not in the document]
Comment (silvano-a) [reply phpstan.neon.dist:2]: I agree. Is there a certain level you propose?

File: src/Commands/AutodiscoveryPackageLockVerify.php (hunk 2 above)
Comment (silvano-a) [reply src/Commands/AutodiscoveryPackageLockVerify.php:49]: I agree, it has currently felt a little bit clunky to me. I'll look into improving the flow!

File: src/Autodiscovery/LaravelPackageManifest.php [hunk 4]
```diff
@@ -24,7 +25,7 @@ public function collectManifestFromComposerAutoload(): Collection
         $manifest = $this->fetchManifest();
 
         if ($manifest->isEmpty()) {
-            throw new \Exception('No packages found in the manifest.');
+            throw new Exception('No packages found in the manifest.');
```
Comment (bonroyage) [inline src/Autodiscovery/LaravelPackageManifest.php:28]: Should we throw a specific exception here, like `AutodisoveryMismatchException`?

**Review by bonroyage — APPROVED** [review]

**Comment by silvano-a** [comment]
@NiekKeijzer Please do let us know your feedback on this PR whenever you see fit. 
I will merge this for now as your feedback has been resolved.

**Comment by NiekKeijzer** [comment]
lgtm

**Pull request merged**